core/timer.go

```diff
@@ -9,11 +9,14 @@ import (
 
 const wakeupTimeout = 30 * time.Second
 
+const wakeupAttempts = 6 // wakeupAttempts is the count of wakeup attempts
```

Comment:
Ich verstehe die Logik nicht. Was passiert wenn ich nur charger oder vehicle wakeup habe? Gibts dann Leerlauf? Werden dann "attempts" ausgelassen?

Reply:
Genau. Ursprünglich hatte ich das anderst. Das war dann von der Logik her aber komplizierter. Das hier war die Bitte von @premultiply, da auch dadurch, wenn es z.B. kein Charger Wakeup gibt, der Charger die Möglichkeit hat sein internen hersteller spezifischen Wakeup durchzuführen, wenn es den dann gibt. Darauf haben wir ja kein Einfluss.

Reply:
Ich glaube meine Antwort war nicht ganz deutlich. "Attempts" werden nicht ausgelassen. Wenn es keine Api gibt, dann passiert bei dem Versuch eben nichts, also sozusagen ein Leerlauf. Es gibt also immer 6 Attempts. Pro Wakeup Typ 3. Und immer abwechselnd. Dabei ist Charger Wakeup immmer als erstes dran. Wenn es z.B. keine CHarger Api gibt, dann hat der Charger noch vor dem ersten Vehicle WakeUp, also 2x 30 Sekunden die Chance einen evnetuellen internen Charger Wakeup durchzuführen über den wir ja keine Handhabe haben. Wie gesagt hat das @premultiply so vorgeschlagen, was ich aber für gut befinde. Vor allem ist der code hier dann viel schlanker.

Reply:
Bekommen wir das noch schöner hin so dass die Anzahl der Versuche auch stimmt? Bei nur einer Methode sind es ja nur 3 und nicht 6?

Reply:
Auch die ggf. internen Weckversuche bzw. -slots des Chargers müssen halt berücksichtigt werden.

Reply:
Mag sein, ist aber eh geraten. Im Zweifel ist die Zahl falsch und zeigt dem User etwas das nicht passiert. Das kriegen wir bestimmt schöner hin.

Reply:
Aber es ist schon richtig, wenn es Charger Wakeup gibt, dann hat die Box definitiv kein internen WakeUp? Wnen ja, hier mein Vorschlag:

- Ich reduzieren die WakeUps auf wirklich als API vorhandenen. Also wenn es nur einen gibt, eben 3 WakeUps. Wenn es keinen gibt 0 ... wie gesagt, so hatte ich es schon.
- Ausnahmen: Wenn es nur einen Vehicle WakeUp gibt, jedoch kein Charger, dann startet der erste WakeUp erst nach 2x30 Skeunden. In dem Fall wäre es ja der einzig existierende WakeUp. So hätte ein evnetuell vorhandener internen Charger Wakeup noch die Change es zu regeln...

@premultiply, @andig, dann das so ok ist für euch, dann versuche ich das so umzusetzten....